src/commands/base-command.mjs

```diff
@@ -420,7 +436,7 @@ export default class BaseCommand extends Command {
     const { buildDir, config, configPath, repositoryRoot, siteInfo } = cachedConfig
     const normalizedConfig = normalizeConfig(config)
     const agent = await getAgent({
-      httpProxy: options.httpProxy,
+      httpProxy: returnDeprecatedOptionValue(options.httpProxy),
```

Comment:
Rather than adding this helper to the code that retrieves the options, do you think we could change this to `options['http-proxy']` and then have the `preAction` hook translate `options.httpProxy` into `options['http-proxy']` until we fully deprecate the camelCase version?

Reply:
I'd rather not mutate the options, as commander by default changes `http-proxy` to `httpProxy`. It's the default for the library to change the option keys to camelcase and I feel like we'd be working against the commander logic 🤔

Reply:
Then I'm not sure why we need to add this helper to every command? If commander translates kebab-case to camelCase, we could continue to use `options.httpProxy` here without any changes? If the idea is to print a deprecation warning, could we use a single `preAction` hook that looks for camelCase options and prints a warning?

Reply:
Because I need to somehow find out if someone is using the old flag or not, which is impossible because of the fact that commander will turn `http-proxy` into `options.httpProxy` I added [this logic](https://github.com/netlify/cli/pull/5395/files#diff-67082469659283bc342bd13f7e20570cb3d1bf5096aad88545a9ac4809d6083bR111) that will _flag_ the old option. 

Like I tried to explain in the summary:

> Why this way? because wether or not the user uses --httpProxy or --http-proxy. The name of the option will always be options.httpProxy 😅. So we need a way to distinguish the old vs the new.

It parses the argument and turns it into an object. That is why we need that helper.

For as far as I was able to figure out, it is not possible for us to check in the `preAction` hook if something is camelCase or not, because commander turns _everything_ into camelcase. Using `--http-proxy` will not create `options[http-proxy]` but `options.httpProxy` :) I hope I was able to explain it a bit better.

Reply:
Oh also: it seems to actually not run `preAction` before running the `init` function here https://github.com/netlify/cli/pull/5395/files#diff-67082469659283bc342bd13f7e20570cb3d1bf5096aad88545a9ac4809d6083bR414

So if we wanted to perform anything in the `preAction` hook, it would already be too late